theseus/optimizer/nonlinear/levenberg_marquardt.py

```diff
@@ -65,3 +65,29 @@ def compute_delta(
             ellipsoidal_damping=ellipsoidal_damping,
             damping_eps=damping_eps,
         )
+
+    def compute_samples(self, n_samples: int = 10, T: float = 1.0) -> torch.Tensor:
```

Comment:
This is identical to the one in gauss newton. Maybe pull this up to nonlinear least squares class instead?

Reply:
Good point, I'll move this up to nonlinear least squares for now. The main consideration is if the same function is applicable for future least-squares solvers that would derive from this class.

Reply:
This proposal is much cleaner so if the logic works generally (say for sparse linear solvers and other possible future solvers) then let's go with this.

Do remember you will need to have run `TheseusLayer.forward` first before calling `TheseusLayer.compute_samples`.

Reply:
On thinking about it more, it does make sense for `compute_samples()` to be independent of the optimizer. I have moved `compute_samples()` to theseus_layer where it uses the current linearization point. However, instead of `linearization`, it is using `linear_solver` as input because we need to compute delta as well within the function.

Reply:
Yes the final computation is a similar functionality as `retract_and_update_variables`.  One important issue to be flagged for the future is to broadcast the `retract_and_update_variables`. It would make sampling much faster.

Reply:
Sounds good, I'll flag an issue for this.